### PR TITLE
Fix compiler errors on Erlang >= 17

### PR DESCRIPTION
**Pull request by bfo**
Since Erlang 18, there is no `erlang:now()` BIF. This change makes the
code use the new `erlang:timestamp()` where appropriate.

Used advice from https://stackoverflow.com/a/34028460.

### CODE REVIEW

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 202 is too long:                                                     handler_state = State1}, Timeout};.

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 200 is too long:                     Timeout = dhcp_package:get_option(ip_address_lease_time, RPkg),.

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 163 is too long:                                                       handler_state = State1}, ?S(To)};.

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 338 is too long:                     {ok, R, State#state{handler_state = S1, last=?CURRENT_TIME}};.

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 160 is too long:                                                     handler_state = State1}, ?S(Timeout)};.

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 340 is too long:                     lager:error("[DHCP] invalid reply package for ~p:~p -> ~p", [M, F, R]).

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 158 is too long:                     Timeout = dhcp_package:get_option(ip_address_lease_time, dhcp_package:get_option(ip_address_lease_time, 3000, RPkg), Pkg),.

File: include/dhcp.hrl
Comment (Licenser) [inline include/dhcp.hrl]: can we make this a function instead of a macro?

**Review by Licenser — CHANGES_REQUESTED** [review]

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 338 is too long:                     {ok, R, State#state{handler_state = S1, last=current_time()}};.

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 202 is too long:                                                     handler_state = State1}, Timeout};.

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 340 is too long:                     lager:error("[DHCP] invalid reply package for ~p:~p -> ~p", [M, F, R]).

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 160 is too long:                                                     handler_state = State1}, ?S(Timeout)};.

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 163 is too long:                                                       handler_state = State1}, ?S(To)};.

**Review by Licenser — APPROVED** [review]

**Comment by Licenser** [comment]
Thanks mate :) looks good, I'll merge once the linter is happy!

File: src/dhcp_fsm.erl
Comment (bfo) [inline src/dhcp_fsm.erl]: Well, how long is too long?

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Line 170 has a tab at column 0.

File: src/dhcp_fsm.erl
Comment (elvisci) [inline src/dhcp_fsm.erl]: Remove the dynamic function call on line 334. Only modules that define callbacks should make dynamic calls.

Comment (bfo) [inline src/dhcp_fsm.erl]: I'm afraid this is beyond me to fix.

Comment (Licenser) [inline src/dhcp_fsm.erl]: no worries it's tricky, I'll fix that after the merge

**Comment by Licenser** [comment]
Just noticed that the elvis errors where old :( sorry for ask you to fix my bad code style, didn't mean to do that. I thought I only had elvis enabled for repos that were clean to begin with.

I cleaned up the rest now and also updated it to use rebar3 and published a hex package.

**Pull request merged**
